subscribe.go

```diff
@@ -0,0 +1,79 @@
+package bps
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+)
+
+var (
+	subReg   = map[string]SubscriberFactory{}
+	subRegMu sync.Mutex
+)
+
+// ----------------------------------------------------------------------------
+
+// SubMessage defines a subscription message details.
+type SubMessage interface {
+	Data() []byte
+}
+
+// Handler defines a message handler.
+type Handler interface {
+	Handle(context.Context, SubMessage) error
+}
+
+// HandlerFunc is a func-based handler adapter.
+type HandlerFunc func(context.Context, SubMessage) error
+
+// Handle handles a single message.
+func (f HandlerFunc) Handle(ctx context.Context, msg SubMessage) error {
+	return f(ctx, msg)
+}
+
+// ----------------------------------------------------------------------------
+
+// Subscriber defines the main subscriber interface.
+type Subscriber interface {
+	// Subscribe subscribes for topic messages and blocks till context is cancelled or error occurs.
+	//
+	// Handler is provided an atomic message batch (entire batch either succeeds or fails).
+	// Batch is guaranteed to contain at least 1 message.
+	Subscribe(ctx context.Context, topic string, handler Handler) error
```

Comment:
Question: probably, better have this? cc @dim 

```go
csm, _ := NewConsumer("kafka://.../namespace?topics=foo,bar,baz")
_ := csm.Subscribe(ctx, handler)
```

Reply:
so I can see why `NewConsumer` should have `ctx` (as this is where we do network IO), not so sure about `Subscribe`. let's not overdo it with context

Reply:
What if we want to dynamically subscribe? Like, to keep subscription not for the whole process run time, but just for some time?

Having context as a "unsubscribe" option would be useful then.

Btw, @dim what about the original question - binding topics in constructor and not in Subscribe? Overthinking?